tests/transform_dialect/cuda/BUILD.bazel

```diff
@@ -71,7 +71,7 @@ iree_lit_test_suite(
         "nomsan",
         "notsan",
         "noubsan",
-        "requires-gpu-nvidia",
+        "requires-gpu-sm80",
```

Comment:
It's fine to have some tests that can only run on a specific architecture when they're arch-specific tests, but we don't want to in general limit our GPU tests to that. Could we separate them out and specifically label them? This is similar to the case of Vulkan fp16

Reply:
Yes, I separated out the tests that require it.

Also, if it helps, requires-gpu-sm80 will have the tests run on anything sm80+, not just sm80 (there's another tag for that - requires-gpu-sm80-only). So it's not limiting to a _single_ architecture, just the earliest that it can run on.